### PR TITLE
Fix: fix Incorrect conversion between integer types in release 1.24

### PR DESCRIPTION
**Pull request by umagnus**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Fix: fix Incorrect conversion between integer types in release 1.24 in GitHub Code Scanning / CodeQL

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: pkg/provider/azure_loadbalancer.go [hunk 1]
```diff
@@ -2173,7 +2173,7 @@ func (az *Cloud) buildHealthProbeRulesForPort(serviceManifest *v1.Service, port
 				}
 			}
 		} else {
-			if port >= 0 || port <= 65535 {
+			if port >= 0 && port <= 65535 {
```
Comment (andyzhangx) [inline pkg/provider/azure_loadbalancer.go:2176]: could you fix the bug in master branch first? thanks.
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/d9b43e25a13be5e01e8eda90584525e07ee8a6ae/pkg/provider/azure_loadbalancer.go#L2176

Comment (umagnus) [reply pkg/provider/azure_loadbalancer.go:2176]: ok, the pr to fix in master branch is https://github.com/kubernetes-sigs/cloud-provider-azure/pull/3339

**Pull request merged**